aggregate/billing_functions/gcp.py

```diff
@@ -0,0 +1,164 @@
+"""
+Cloud function that runs {TBA} that synchronises a portion of data from:
+
+FROM:   billing-admin-290403.billing.gcp_billing_export_v1_01D012_20A6A2_CBD343
+TO:     billing-admin-290403.billing_aggregate.aggregate
+
+Tasks:
+
+- Needs to convert {billing_project_id} into DATSET
+- Only want to transfer data from the projects in the server-config
+- Can't duplicate rows, so:
+    - just grab only settled data within START + END of previous time period
+- Service ID should be faithfully handed over
+- Should search and update for [START_PERIOD, END_PERIOD)
+"""
+
+import json
+import hashlib
+
+from datetime import datetime
+from typing import Dict
+
+from pandas import DataFrame
+import google.cloud.bigquery as bq
+
+try:
+    from . import utils
+except ImportError:
+    import utils
+
+
+logger = utils.logger
+
+
+##########################
+#    INPUT PROCESSORS    #
+##########################
+
+
+def from_request(request):
+    """
+    From request object, get start and end time if present
+    """
+    start, end = utils.get_start_and_end_from_request(request)
+    main(start, end)
+
+
+def from_pubsub(data=None, _=None):
+    """
+    From pubsub message, get start and end time if present
+    """
+    start, end = utils.get_start_and_end_from_data(data)
+    main(start, end)
+
+
+#################
+#    MIGRATE    #
+#################
+
+
+def migrate_billing_data(start, end, dataset_to_gcp_map) -> int:
+    """
+    Get the billing date in the time period
```

Comment:
As a general comment, function descriptions should be passive ("_Gets_ the billing date...") instead of a command ("_Get_ the billing date...").